nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -2772,4 +2793,92 @@ mod test {
         db.cleanup().await.unwrap();
         logctx.cleanup_successful();
     }
+
+    /// Creates a representative collection, deletes it, and walks through
+    /// tables to ensure that the subcomponents of the inventory have been
+    /// deleted.
+    ///
+    /// NOTE: This test depends on the naming convention "inv_" prefix name
+    /// to identify pieces of the inventory.
+    #[tokio::test]
+    async fn test_inventory_deletion() {
```

Comment:
This is a new test I added, out of paranoia, to try to help catch future cases like this, if any "inv_..." tables are added in the future, and they are not deleted.

Reply:
This might fix #5305

Reply:
Yeah, it's not exactly a database snapshot, but querying for all present and future `inv_` tables is an approximation

Reply:
Yeah the ticket isn't so much about that specific implementation so much as having a test to catch this problem.

I can't think of why you'd need a snapshot.  A possible risk is that another inventory collection is inserted by a background task in the meantime, but since this is a datastore test and not an app-level test, there's no Nexus, and so that should be impossible.

Do you think we should have a test for the representative collection that it has rows for all `inv_*` tables?  I'm a little afraid this test won't catch this problem in the future because someone might forget to add it to that, too.  (It wouldn't have caught this specific instance for that reason, right?)

Reply:
I'm gonna merge this PR first, but yes, totally agreed, we should have a test for that. I'll file another issue.

Reply:
https://github.com/oxidecomputer/omicron/issues/6636